core/deployment/src/main/java/io/quarkus/deployment/pkg/builditem/DockerBuild.java

```diff
@@ -0,0 +1,21 @@
+
+package io.quarkus.deployment.pkg.builditem;
+
+import java.util.function.BooleanSupplier;
+
+import io.quarkus.deployment.pkg.ContainerConfig;
+import io.quarkus.deployment.util.ExecUtil;
+
+public class DockerBuild implements BooleanSupplier {
+
+    private final ContainerConfig containerConfig;
+
+    DockerBuild(ContainerConfig containerConfig) {
+        this.containerConfig = containerConfig;
+    }
+
+    @Override
+    public boolean getAsBoolean() {
+        return containerConfig.build && ExecUtil.exec("docker", "version");
```

Comment:
This should maybe re-use io.quarkus.deployment.pkg.PackageConfig#type, and add docker-jvm and docker-native as possible output targets. 

This was the original intent anyway (and is why the package type is not an enum).

Reply:
In terms of code it doesn't really make any real difference (doesn't make code cleaner, smarter etc). So, the only difference that it makes is the actual way to configure it.

So, with your suggestion, the user would have to:

    mvn clean package -Dquarkus.package.type=docker-jvm

with the existing code, the user woruld have to use:

    mvn clean package -Dquarkus.container.build=true


Of course, this is something subjective, however the existing approach rhymes better with:

     mvn clean package -Dquarkus.container.deploy=true

So, I'd like to suggest reconsidering this point.